mantidproject/mantid
# Add tab/widget to select region of interest etc. from stacks of images

Fixes #13140, adding the "ROI etc." tab in the Tomo custom interface.

**To test**:
- The new widget and helper classes come with unit tests that should pass.
- Code should be decently organized.  The new widget lives inside the directory of the tomo interface, as it is a very specific widget.
- For a manual/interactive test I can suggest the following steps (I can show you quickly how to use the interface):
  - Custom Interfaces -> Diffraction -> Tomographic Reconstruction
  - change to the "ROI etc." tab
  - the panel on the right should be disabled
  - if you try to browse/open a wrong directory it should show you a warning pop-up
  - to open a proper stack of images use this example (ISIS babylon5): \olympic\Babylon5\Public\fedemp\test_stack_phantom_easter. Note: in the widget, browse and select that root directory which contains three subdirectories data_..., dark_..., flat_...).
  - see that:
  - you can scroll through the images of the stack
  - you can select the center, region of interest (ROI), and area of normalization:
    - using the spin boxes for fine precision/friends of the keyboard
    - using the "select" buttons and then clicking and dragging with the mouse on the image in order to select a point (center) or a rectangle (ROI and area of normalization)
  - the documentation about this should read acceptable, even though is far from complete at this stage
  - the selection should be kept when you slide through the images

Release notes: this is one first step in a series of many additions that will be added to the release notes later on.
